packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncherItem.tsx

```diff
@@ -25,29 +27,57 @@ export interface ApplicationLauncherItemProps {
    * } />
    */
   component?: React.ReactNode;
+  /** Flag indicating if the item is favorited */
+  isFavorite?: boolean;
+  /** ID of the item. Required for tracking favorites. */
+  id?: string;
 }
 
 export const ApplicationLauncherItem: React.FunctionComponent<ApplicationLauncherItemProps & DropdownItemProps> = ({
   className = '',
+  id,
   children,
   icon = null,
   isExternal = false,
   href,
   tooltip = null,
   tooltipProps = null,
   component = 'a',
+  isFavorite = false,
   ...props
 }: ApplicationLauncherItemProps & DropdownItemProps) => (
   <ApplicationLauncherItemContext.Provider value={{ isExternal, icon }}>
-    <DropdownItem
-      component={component}
-      href={href || null}
-      className={css(isExternal && styles.modifiers.external, className)}
-      tooltip={tooltip}
-      tooltipProps={tooltipProps}
-      {...props}
-    >
-      {children && <ApplicationLauncherContent>{children}</ApplicationLauncherContent>}
-    </DropdownItem>
+    <ApplicationLauncherContext.Consumer>
+      {({ onFavorite }) => (
+        <DropdownItem
+          id={id}
+          component={component}
+          href={href || null}
+          className={css(isExternal && styles.modifiers.link, className)}
```

Comment:
Oh sorry this probably wasn't clear from the implementation. Prior to this PR, the app launcher item (`.pf-c-app-launcher__menu-item`) is either a button or a link (some sort of interactive element you click on and it does the thing). In this PR, we introduce the ability to have multiple interactive things (items) inside of a single main row/item in the menu. To enable this (for example, when an item in the menu is favoritable, there are 2 individual interactive items in the main menu item), we allow this by wrapping the individual items with `.pf-c-app-launcher__menu-wrapper`, then placing the individual items in the wrapper. Those individual items will need classes to handle the spacing. The main item (the button/link) will use the modifier `.pf-c-app-launcher__menu-item.pf-m-link`, and any actions to the right of it will use `.pf-c-app-launcher__menu-item.pf-m-action`.

Which is a long way of saying that all of the buttons/links in a favoritable item need `styles.modifiers.link`, not just external links.